src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -30,574 +30,534 @@
+                                      final BlockState originalBlock) {
+        block.setType(getMaterial());
+        BlockState state = block.getState();
+        final SkullActionData s = (SkullActionData) blockActionData;
+
+        if (state.getBlockData() instanceof Rotatable) {
+            final Rotatable r = (Rotatable) state.getBlockData();
+            r.setRotation(s.getRotation());
+            state.setBlockData(r);
+        } else {
+            final Directional d = (Directional) state.getBlockData();
+            d.setFacing(s.getRotation());
+            state.setBlockData(d);
+        }
+        state = block.getState();
+
+        if (!s.owner.isEmpty()) {
+            final Skull skull = (Skull) state;
+            skull.setOwningPlayer(Bukkit.getOfflinePlayer(EntityUtils.uuidOf((s.owner))));
+        }
+        BlockStateChange stateChange = new BlockStateChange(originalBlock, state);
+        return new ChangeResult(ChangeResultType.APPLIED, stateChange);
+    }
+
+    ChangeResult removeBlock(Player player, QueryParameters parameters, boolean isPreview, Block block) {
+
+        BlockStateChange stateChange;
+
+        if (!block.getType().equals(Material.AIR)) {
+
+            // Ensure it's acceptable to remove the current block
+            if (!BlockUtils.isAcceptableForBlockPlace(block.getType())
+                    && !BlockUtils.areBlockIdsSameCoreItem(block.getType(), getMaterial())
+                    && !parameters.hasFlag(Flag.OVERWRITE)) {
+                return new ChangeResult(ChangeResultType.SKIPPED, null);
+            }
+
+            if (!isPreview) {
+
+                // Capture the block before we change it
+                final BlockState originalBlock = block.getState();
+
+                // Set
+                block.setType(Material.AIR);
+
+                // Capture the new state
+                final BlockState newBlock = block.getState();
+
+                // Store the state change
+                stateChange = new BlockStateChange(originalBlock, newBlock);
+
+            } else {
+
+                // Otherwise, save the state so we can cancel if needed
+                final BlockState originalBlock = block.getState();
+                // Note: we save the original state as both old/new so we can
+                // re-use blockStateChanges
+                stateChange = new BlockStateChange(originalBlock, originalBlock);
+
+                // Preview it
+                EntityUtils.sendBlockChange(player, block.getLocation(), Bukkit.createBlockData(Material.AIR));
+
+                // Send preview to shared players
+                for (final CommandSender sharedPlayer : parameters.getSharedPlayers()) {
+                    if (sharedPlayer instanceof Player) {
+                        EntityUtils.sendBlockChange((Player) sharedPlayer, block.getLocation(),
+                                Bukkit.createBlockData(Material.AIR));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Material.AIR' due to existing static import 'org.bukkit.Material.*'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5446892)